src/Symfony/Component/Messenger/Command/AbstractFailedMessagesCommand.php

```diff
@@ -29,20 +30,25 @@
  */
 abstract class AbstractFailedMessagesCommand extends Command
 {
-    private $receiverName;
-    private $receiver;
+    private $globalFailureReceiverName;
+    private $failureTransports;
 
-    public function __construct(string $receiverName, ReceiverInterface $receiver)
+    public function __construct(?string $globalFailureReceiverName, $failureTransports)
     {
-        $this->receiverName = $receiverName;
-        $this->receiver = $receiver;
+        $this->failureTransports = $failureTransports;
+        $this->globalFailureReceiverName = $globalFailureReceiverName;
+        if (!$failureTransports instanceof ServiceLocator) {
+            trigger_deprecation('symfony/messenger', '5.2', 'Passing failureTransports should now pass a ServiceLocator', __METHOD__);
+
+            $this->failureTransports = new ServiceLocator([$this->globalFailureReceiverName => function () use ($failureTransports) { return $failureTransports; }]);
+        }
 
         parent::__construct();
     }
 
-    protected function getReceiverName(): string
```

Comment:
This is a BC break. We need to leave the old method but trigger a deprecation in it.